app/main.py

```diff
@@ -1 +1,17 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import VaccineError, NotWearingMaskError
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    masks_to_buy = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            return "All friends should be vaccinated"
```

Comment:
The current logic returns immediately when a `VaccineError` is encountered, which means it will not check the rest of the friends. If the intention is to ensure all friends are vaccinated before proceeding, this is correct. Otherwise, consider accumulating errors and reporting them after checking all friends.